python/ray/data/_internal/numpy_support.py

```diff
@@ -26,10 +26,13 @@ def is_valid_udf_return(udf_return_col: Any) -> bool:
 
 def is_scalar_list(udf_return_col: Any) -> bool:
     """Check whether a UDF column is is a scalar list."""
-
-    return isinstance(udf_return_col, list) and (
-        not udf_return_col or np.isscalar(udf_return_col[0])
-    )
+    if not (isinstance(udf_return_col, (list, np.ndarray))):
+        return False
+    if len(udf_return_col) == 0:
+        return True
+    if np.isscalar(udf_return_col[0]):
+        return True
+    return is_scalar_list(udf_return_col[0])
```

Comment:
What's meant by "scalar list" here? I expect it to mean a `list[float]` like `[1, 2, 3]` and not a nested arrays like `[[1], [2]]`, but with this implementation nested lists would be considered scalar lists.

Reply:
why we only need to check `udf_return_col[0]`?

Reply:
@bveeramani @c21 I've updated the PR with a new way to fix. I.e, do not convert each individual element, as long as the whole list is a nested list.